vault/activity_log_test.go

```diff
@@ -4692,16 +4692,31 @@ func newMockTimeNowClock(startAt time.Time) timeutil.Clock {
 	return &mockTimeNowClock{start: startAt, created: time.Now()}
 }
 
+// NewTimer returns a timer with a channel that will return the correct time,
+// relative to the starting time
+func (m mockTimeNowClock) NewTimer(d time.Duration) *time.Timer {
```

Comment:
added to the function description, but it's needed because `activeFragmentWorker` uses the returned value from the timer channel. If this function didn't exist, then the precomputed queries would end up getting written with an end time of the actual time.Now(), rather than the mocked time